### PR TITLE
feat(charts): Give the possibility to add labels on more resources

### PR DESCRIPTION
**Pull request by elagree**
Currently it's possible to add labels on Jobs but not on Pods created by the Job. Let's allow it.

Same problem on operator subchart for deployments and pod created by the deployment.

### CODE REVIEW

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 1]
```diff
@@ -655,6 +655,63 @@ the same Kubernetes cluster or installing the CRDs on your own you should not ha
...
+| `object` | `{}`          |
+
+[Kubernetes reference](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+
+Kubernetes annotations which should be applied to the `Deployment` created by the chart.
+
+`values.yaml` example:
+
+  ```yaml
+    operator:
```
Comment (tigrato) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:671]: Can you reduce the identation here?

**Review by tigrato — APPROVED** [review]
This is a great submission. 

Thanks for that. Once you address my only NIT, we can merge it.

File: docs/pages/reference/helm-reference/teleport-cluster.mdx (hunk 1 above)
Comment (elagree) [reply docs/pages/reference/helm-reference/teleport-cluster.mdx]: Hello @tigrato 
Thanks for your feedback. Is that better now ?
Thank you

Comment (tigrato) [reply docs/pages/reference/helm-reference/teleport-cluster.mdx]: It's perfect.

**Comment by tigrato** [comment]
Hi @elagree,

Thanks for your contribution. We are currently unable to run our tests on
external contributions. As such, I'll be helping you validate and
merge your pull request. I've created an internal PR here:

https://github.com/gravitational/teleport/pull/49624

Please wait for another approver and I'll get your
code merged. You will retain authorship of your commits.

**Pull request closed without merging**